wasm/jit/jit_value_location_amd64.go

```diff
@@ -155,6 +156,9 @@ func (s *valueLocationStack) push(loc *valueLocation) {
 	} else {
 		s.stack[s.sp] = loc
 	}
+	if s.sp > s.maxHeight {
```

Comment:
unrelated but would this condition trigger when `s.sp >= uint64(len(s.stack))` if so, maybe move the resetting `s.maxHeight = s.sp` to there? unless we are expecting shrinking?

Reply:
nope, we have to track s.sp whenever we `push`, meaning we advance sp😄  `s.sp >= uint64(len(s.stack)` satisfies when we try pushing more items than the current slice can accommodate.